### PR TITLE
feat: Add helm repository and service yaml for Nutanix AI Endpoint

### PR DESCRIPTION
**Pull request by johnliu25**
Add helm repositories and services similar to https://github.com/mesosphere/dkp-catalog-applications/tree/main/services/kafka-operator

Will create another separate repo and check in this PR. Will not merge to dkp-catalog-applications branch.

### CODE REVIEW

**Comment by deepakm-ntnx** [comment]
you can also create a catalog git repo using a private repo. <pre>
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: GitRepository
metadata:
  name: demo-repo
  namespace: demo
spec:
  interval: 1m0s
  ref:
    branch: main
  secretRef:
    name: <secret name>
  url: https://github.com/<org or my user>/<repo>
</pre>

we dont want to make this public yet. pls remove this PR and move the changes to a private repo which can be created in your github account as private repo

File: services/nutanix-ai/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+displayName: Nutanix AI
+description: NAI core components.
+allowMultipleInstances: false
+category:
+  - Data Services
```
Comment (deepakm-ntnx) [inline services/nutanix-ai/metadata.yaml:5]: Artificial Intelligence would be a better category name?

File: services/nutanix-ai/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+certifications:
+  - certified
```
Comment (deepakm-ntnx) [inline services/nutanix-ai/metadata.yaml:7]: pls see more names here https://docs.google.com/document/d/1TFvMb8xlEckWDzt4DrVhqENEiwAIefkxTt2loY2LL3Q/edit#bookmark=id.kdal4251v7wo
we should add supported too

File: services/nutanix-ai/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+displayName: Nutanix AI
```
Comment (deepakm-ntnx) [inline services/nutanix-ai/metadata.yaml:1]: pls add type: catalog per https://docs.google.com/document/d/1TFvMb8xlEckWDzt4DrVhqENEiwAIefkxTt2loY2LL3Q/edit#bookmark=id.kdal4251v7wo

File: services/nutanix-ai/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+description: NAI core components.
```
Comment (deepakm-ntnx) [inline services/nutanix-ai/metadata.yaml:2]: please add dependencies to show the list of dependent apps from catalog

File: services/nutanix-ai/metadata.yaml (hunk 1 above)
Comment (johnliu25) [reply services/nutanix-ai/metadata.yaml]: I think it is a better name.

File: services/nutanix-ai/metadata.yaml (hunk 2 above)
Comment (johnliu25) [reply services/nutanix-ai/metadata.yaml:7]: yes

File: services/nutanix-ai/metadata.yaml (hunk 3 above)
Comment (johnliu25) [reply services/nutanix-ai/metadata.yaml:1]: the type has been added.

File: services/nutanix-ai/metadata.yaml (hunk 4 above)
Comment (johnliu25) [reply services/nutanix-ai/metadata.yaml:2]: updated

File: services/nutanix-ai/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+category:
+  - Artificial Intelligence
+certifications:
+  - certified
+  - supported
+  - airgapped
+licensing:
+  - Ultimate
+  - Enterprise
+requiredDependencies:
```
Comment (deepakm-ntnx) [inline services/nutanix-ai/metadata.yaml:14]: this should be dependencies and not requiredDependencies. which did get u this string?

Comment (johnliu25) [reply services/nutanix-ai/metadata.yaml:16]: https://github.com/mesosphere/kommander-applications/blob/main/services/nkp-insights/metadata.yaml#L9

**Comment by johnliu25** [comment]
@deepakm-ntnx could we created a private repo under nutanix for now? I have tried yesterday and authentication issue need to hbe solved in this case.
```
john@nkp-john:~$ kubectl get gitrepository -n john-catalog-7slm4-bhnbp --kubeconfig="/home/john/john-dev.conf"
NAME                   URL                                                     AGE    READY   STATUS
nutanix-apps-catalog   https://github.com/johnliu25/dkp-catalog-applications   116s   False   failed to checkout and determine revision: unable to clone 'https://github.com/johnliu25/dkp-catalog-applications': authentication required
```

File: helm-repositories/ntnx-charts/ntnx-charts.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: nutanix.github.io/helm-releases
```
Comment (gracedo) [inline helm-repositories/ntnx-charts/ntnx-charts.yaml:5]:
```suggestion
  name: nutanix.github.io-helm-releases
```

File: services/nutanix-ai/1.0.0/nutanix-ai.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+metadata:
+  name: nai-core
+  namespace: ${releaseNamespace}
+spec:
+  chart:
+    spec:
+      chart: nai-core
+      sourceRef:
+        kind: HelmRepository
+        name: nutanix.github.io/helm-releases
```
Comment (gracedo) [inline services/nutanix-ai/1.0.0/nutanix-ai.yaml:13]:
```suggestion
        name: nutanix.github.io-helm-releases
```

**Review by gracedo — COMMENTED** [review]
To resolve
```
HelmRepository/kommander/nutanix.github.io/helm-releases dry-run failed: invalid resource name "nutanix.github.io/helm-releases": [may not contain '/']...
```

File: services/nutanix-ai/metadata.yaml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+category:
+  - Artificial Intelligence
+certifications:
+  - certified
+  - supported
+  - airgapped
+licensing:
+  - Ultimate
+  - Enterprise
+  - Pro
```
Comment (deepakm-ntnx) [inline services/nutanix-ai/metadata.yaml:14]: per PRD, only Pro and Ultimate needs to be supported/

**Pull request closed without merging**